input/system/selfhosted/otel_handler.go

```diff
@@ -0,0 +1,210 @@
+// 3. jsonlog wrapped in K8s context (via fluentbit) as OTel key/value map
+//
+// Other variants (e.g. csvlog, or plain messages in a K8s context) are currently
+// not supported and will be ignored.
+
+var k8sSelectorRegexp = regexp.MustCompile(`\s*([^!=\s]+)\s*([!=]+)\s*([^\s]+)\s*`)
+
+func logLineFromJsonlog(record *common.KeyValueList, tz *time.Location) (state.LogLine, *state.LogLine) {
+	var logLine state.LogLine
+	logLine.CollectedAt = time.Now()
+	logLine.UUID = uuid.NewV4()
+
+	// If a DETAIL line is set, we need to create an additional log line
+	detailLineContent := ""
+
+	for _, rv := range record.Values {
+		if rv.Key == "log_time" {
+			logLine.OccurredAt = logs.GetOccurredAt(rv.Value.GetStringValue(), tz, false)
+		}
+		if rv.Key == "user_name" {
+			logLine.Username = rv.Value.GetStringValue()
+		}
+		if rv.Key == "database_name" {
+			logLine.Database = rv.Value.GetStringValue()
+		}
+		if rv.Key == "process_id" {
+			backendPid, _ := strconv.ParseInt(rv.Value.GetStringValue(), 10, 32)
+			logLine.BackendPid = int32(backendPid)
+		}
+		if rv.Key == "application_name" {
+			logLine.Application = rv.Value.GetStringValue()
+		}
+		if rv.Key == "session_line_num" {
+			logLineNumber, _ := strconv.ParseInt(rv.Value.GetStringValue(), 10, 32)
+			logLine.LogLineNumber = int32(logLineNumber)
+		}
+		if rv.Key == "message" {
+			logLine.Content = rv.Value.GetStringValue()
+		}
+		if rv.Key == "detail" {
+			detailLineContent = rv.Value.GetStringValue()
+		}
+		if rv.Key == "error_severity" {
+			logLine.LogLevel = pganalyze_collector.LogLineInformation_LogLevel(pganalyze_collector.LogLineInformation_LogLevel_value[rv.Value.GetStringValue()])
+		}
+	}
+	if detailLineContent != "" {
+		detailLine := logLine
+		detailLine.Content = detailLineContent
+		detailLine.LogLevel = pganalyze_collector.LogLineInformation_DETAIL
+		return logLine, &detailLine
+	}
+
+	return logLine, nil
+}
+
+func skipDueToK8sFilter(kubernetes *common.KeyValueList, server *state.Server, prefixedLogger *util.Logger) bool {
+	var k8sPodName string
+	var k8sNamespaceName string
+
+	k8sLabels := make(map[string]string)
+	for _, rv := range kubernetes.Values {
+		if rv.Key == "pod_name" {
+			k8sPodName = rv.Value.GetStringValue()
+		}
+		if rv.Key == "namespace_name" {
+			k8sNamespaceName = rv.Value.GetStringValue()
+		}
+		if rv.Key == "labels" {
+			for _, ll := range rv.Value.GetKvlistValue().Values {
+				k8sLabels[ll.Key] = ll.Value.GetStringValue()
+			}
+		}
+	}
+
+	if server.Config.LogOtelK8SPod != "" {
+		parts := strings.SplitN(server.Config.LogOtelK8SPod, "/", 2)
+		if len(parts) == 2 {
+			if k8sNamespaceName != parts[0] || k8sPodName != parts[1] {
+				return true
+			}
+		} else if len(parts) == 1 {
+			if k8sPodName != parts[0] {
+				return true
+			}
+		} else {
+			prefixedLogger.PrintWarning("Pod specification for OTel server not valid (need zero or one / separator): \"%s\", skipping log record\n", server.Config.LogOtelK8SPod)
```

Comment:
Maybe we should put this in `config/read.go`? It seems silly to warn about this for every log line. I think we could also consider erroring out on this at startup, no?

Reply:
Maybe we should also `SplitN` this while parsing, and set values on Config that don't correspond to config settings directly (one for name, one for namespace)? I doubt it's a bottleneck here, but that might be easier to follow. Lines 102-114 could become

```golang
if server.Config.LogOtelK8SPod != "" {
  if server.Config.LogOtelK8SPodNamespace != "" && server.Config.LogOtelK8SPodNamespace != k8sNamespaceName {
    return true
  }
  if server.Config.LogOtelK8SPod != k8sPodName {
    return true
  }
}
```

Reply:
Thanks for the idea! I refactored this, and also added the test for the label/selector matching part as it was actually in TODO of the original PR. Ideally we want to use the k8s code for this I'd say, but that'll involve lots of pulling deps so I decided not to do that at least in this PR.

Reply:
Did you push up those changes? I don't see them in this branch.

Reply:
Wow, I swear I did update this (well I did everything to prepare this, which is already pushed), but I don't see them too 🙃 Now I'm modifying, maybe I did in fact forget to update this part 🤦 Thanks for noticing! Update pushed 👍